### PR TITLE
updated old VSO build agent template to work with VSTS

### PR DESCRIPTION
**Pull request by Geertvdc**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checkist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* updated VSO naming to VSTS
* updated powershell script to work with new VSTS apis instead of no longer working VSO apis

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Geertvdc__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 1]
```diff
@@ -67,28 +67,22 @@
         "description": "Globally unique name for per region for the public IP address. For instance, myVMuniqueIP.westus.cloudapp.azure.com. It must conform to the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
       }
     },
-    "vsoAccount": {
+    "vstsAccount": {
       "type": "string",
       "metadata": {
-        "description": "The Visual Studio Online account name, that is, the first part of yourvsoAccount.visualstudio.com"
+        "description": "The Visual Studio Team Services account name, that is, the first part of yourvstsAccount.visualstudio.com"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: typo - "your vstsAccount.visualstudio.com" - space between your and the uri

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 2]
```diff
@@ -67,28 +67,22 @@
...
       "metadata": {
-        "description": "The Visual Studio Online password configured as Alternate Authentication Credentials."
+        "description": "The personal access token to connect to VSTS"
       }
     },
     "poolName": {
       "type": "string",
       "metadata": {
-        "description": "The Visual Studio Online build agent pool for this build agent to join. Use 'Default' if you don't have a separate pool."
+        "description": "The Visual Studio Team Services build agent pool for this build agent to join. Use 'Default' if you don't have a separate pool."
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: add a default value here of "Default"?

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 3]
```diff
@@ -235,9 +229,9 @@
             "typeHandlerVersion": "1.4",
             "settings": {
               "fileUris": [
-                "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/visual-studio-vsobuildagent-vm/InstallVSOAgent.ps1"
+                "https://raw.githubusercontent.com/geertvdc/azure-quickstart-templates/master/visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: See BP checklist for how to use artifacts in a sample

**Comment by Geertvdc** [comment]
@bmoore-msft is there anything else that's blocking the pull request from being merged? i think i fixed all your comments about a week ago.?

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 4]
```diff
@@ -235,9 +230,9 @@
             "typeHandlerVersion": "1.4",
             "settings": {
               "fileUris": [
-                "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/visual-studio-vsobuildagent-vm/InstallVSOAgent.ps1"
+                "https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: See the BP guide for handling artifacts in samples...

File: visual-studio-vstsbuildagent-vm/azuredeploy.parameters.json [hunk 5]
```diff
@@ -18,22 +18,19 @@
       "value": "GEN-UNIQUE"
     },
     "vmSize": {
-      "value": "Standard_DS2"
+      "value": "Standard_DS1"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.parameters.json]: put defaultValues in the template and remove other values not needed by CI (so the use is prompted for relevant values.

**Pull request merged**